CodeQL.ps1

```diff
@@ -0,0 +1,195 @@
+function Install-Gdn {
```

Comment:
Is this script coming from somewhere else or is this the first time it's appeared in our infra?

Reply:
It's new. I wrote it for this and will use it in the other non-arcade repos that need CodeQL. 

Also, down the priority list, there an Operations issue to always use the latest Guardian. I will probably use these ideas (like how `Install-Gdn` works) to accomplish that.

Reply:
I am a little concerned about having two separate pieces of infrastructure for interacting with guardian, especially since this exists outside of arcade and thus can't be easily maintained as part of arcade updates. Do you have an idea for what maintaining/updating this script looks like? I sort of feel like it would be easier to keep it as part of arcade and just force repos to onboard as part of the security push, honestly.

The actual code here LGTM tho, and honestly we should consider using some of it to overhaul our guardian interactions in arcade.

Reply:
> I am a little concerned about having two separate pieces of infrastructure for interacting with guardian, especially since this exists outside of arcade and thus can't be easily maintained as part of arcade updates.

Certainly agree. The initial plan after chatting with Matt was that we needed "some way" to get CodeQL coverage on (1) non-arcade repos and (2) non-latest arcade repos. After an aborted trial with the SDL Extension, this was my answer. 

I do also think it has good bones to influence the current Arcade SDL implementation (which does make CodeQL use awkward). I think that may come with dotnet/core-eng#14160.

Reply:
> Do you have an idea for what maintaining/updating this script looks like? 

I expect this will be low-to-no maintenance. Arcade's SDL doesn't need to be changed very often, and this is simpler. And, it's more of a facade, so any changes be straightforward with however Guardian ends up changing.